### DISCUSSION TITLE
Generic async file reader

### DISCUSSION THREAD

**Discussion by laultman** [StereoKit/StereoKit#760] [Q&A] [answered]
@maluoi We are loading files and stuff constantly in the HoloLens as dynamic items. I was looking at the C++ code for your file load routines. We only need text and binary files loaded. Right now, we are using the Windows UWP runtime to do this, but it takes a lot of plumbing.
Couldn't you have a file loader that reads/writes to/from an arbitrary folder location that is native to SK that returns either a text or bin file? Even better if it were a "Step" that doesn't draw but still gets updated in the Stepper. I'm thinking of an event driven model that uses the Step to let us fire and handle events but not interrupt the drawing steps.
I need to save data that I am capturing in the 3D space (sort of like a point cloud) and I need to have JSON data and binary's that I encoded/decoded from external systems. Our customers are large enterprise clients that have a host of data sources to contend with. We are standardizing as much of it as possible. All this is ultimately landing in systems like SharePoint, Dynamics 365, Oracle, and custom applications.

**Comment by maluoi** [comment, marked as answer]
StereoKit does have `Platform.ReadFileBytes` and `Platform.ReadFileText` which you could use with `Task.Run`, but this is mostly useful for projects using SK's C API. C#'s own file loading APIs are likely to be much better for you in both simple and complex cases. Adding async file loading APIs directly to SK would likely be out-of-scope, unweildly and error prone, especially since SK's focus is on being an immediate mode API.

I'd recommend sticking to C# standard file IO APIs, or UWPs if you really need it.

**Reply by laultman** [reply]
Thanks, good to know!
